text/src/autogluon/text/automm/data/collator.py

```diff
@@ -0,0 +1,324 @@
+            ret.append(ele_fn([ele[i] for ele in data]))
+        return tuple(ret)
+
+
+class List:
+    r"""Simply forward the list of input data.
+
+    This is particularly useful when the Dataset contains textual data
+    and in conjunction with the `Tuple` collate function.
+
+    """
+
+    def __call__(self, data):
+        """
+        Arguments:
+            data (list): The list of samples
+
+        Returns:
+            ret (list): The input list
+        """
+        return list(data)
+
+
+class Dict:
+    r"""Wrap multiple collate functions together and apply it to merge inputs from a dict.
+
+    The generated batch samples are stored as a dict with the same keywords.
+
+    Each data sample should be a dict and the fn corresponds to `key` will be applied on the
+    input with the keyword `key`.
+    For example, each data sample is {'data': nd_data, 'label': nd_label}.
+    You can merge the data and labels using
+    `Dict({'data': DataCollate, 'label': LabelCollate})` to collate the nd_data and nd_label.
+
+    Arguments:
+        fn_dict (dict): A dictionary that contains the key-->collate function mapping.
+
+    """
+
+    def __init__(self, fn_dict):
+        self._fn_dict = fn_dict
+        if not isinstance(fn_dict, dict):
+            raise ValueError("Input must be a dictionary! type of input={}".format(type(fn_dict)))
+        for fn in fn_dict.values():
+            if not hasattr(fn, "__call__"):
+                raise ValueError("Elements of the dictionary must be callable!")
+        self._fn_dict = fn_dict
+
+    def __call__(self, data):
+        """
+
+        Arguments:
+            data (dict): The samples to collate. Each sample should be a dictionary
+
+        Returns:
+            ret (dict): The resulting dictionary that stores the merged samples.
+        """
+        ret = dict()
+        for k, ele_fn in self._fn_dict.items():
+            ret[k] = ele_fn([ele[k] for ele in data])
+        return ret
+
+
+class PadImage:
+    def __call__(self, image_list):
+        """
+        Arguments:
+            image_list (list): The list of images
+
+        Returns:
+            ret (tensor): The torch tensor
+        """
+        batch_size = len(image_list)
+        img_sizes = [img.shape for img in image_list]
+        for size in img_sizes:
+            assert (len(size) == 3 and size[0] == 3),\
+                f"Collate error, an image should be in shape of (3, H, W), instead of given {size}"
+        max_height = max([i[1] for i in img_sizes])
+        max_width = max([i[2] for i in img_sizes])
+
+        ret = torch.zeros(batch_size, 3, max_height, max_width,
+                          device=image_list[0].device,
+                          dtype=image_list[0].dtype)
+
+        for i, image in enumerate(image_list):
+            # assign to the top-left since convolution starts processing from top-left
+            ret[i, :, :image.shape[1], :image.shape[2]] = image
+
+        return ret
```

Comment:
1. Please reuse torch code: [`torchvision.transforms.Pad()`](https://pytorch.org/vision/stable/transforms.html) or pillow version (that one I think is faster).
2. Zero padding in many cases not the optimal choice.

Reply:
@zhiqiangdon I think we may just remove PadImage since the functionality is supported by `Pad()`.

Reply:
PadImage removed.